File: backend/app/schemas/pricing.py
"""Public pricing catalog schemas."""

from datetime import datetime
from typing import Literal, Optional

from pydantic import BaseModel, Field

from app.schemas.admin import PricingProviderMappings


class PublicPricingPlan(BaseModel):
    plan_key: Literal["free", "pro_monthly", "pro_yearly", "enterprise"]
    display_name: str
    is_public: bool
    price_amount_cents: int
    display_price: str
    currency: str
    billing_interval: str
    description: Optional[str] = None
    provider_mappings: PricingProviderMappings = Field(default_factory=PricingProviderMappings)
    sort_order: int
    highlighted: bool
    updated_at: datetime | None = None


class PublicPricingPlanList(BaseModel):
    source: str
    plans: list[PublicPricingPlan]

File: src/services/api/pricing.ts
import apiClient from '../http'

export type PricingPlanKey = 'free' | 'pro_monthly' | 'pro_yearly' | 'enterprise'

export interface PricingProviderMappings {
  stripe: {
    price_id: string
  }
  paypal: {
    plan_id: string
    product_id: string
  }
  gmpay: {
    amount_cents: number
    currency: string
    token: string
    network: string
  }
}

export interface PublicPricingPlan {
  plan_key: PricingPlanKey
  display_name: string
  is_public: boolean
  price_amount_cents: number
  display_price: string
  currency: string
  billing_interval: 'none' | 'month' | 'year' | 'custom' | string
  description: string | null
  provider_mappings: PricingProviderMappings
  sort_order: number
  highlighted: boolean
  updated_at?: string | null
}

export interface PublicPricingPlanList {
  source: string
  plans: PublicPricingPlan[]
}

export const pricingAPI = {
  async listPlans(): Promise<PublicPricingPlanList> {
    const response = await apiClient.get<PublicPricingPlanList>('/api/v1/pricing/plans')
    return response.data
  },
}
